src/server.c

```diff
@@ -3445,15 +3451,17 @@ void call(client *c, int flags) {
 
     /* populate the per-command statistics that we show in INFO commandstats. */
     if (flags & CMD_CALL_STATS) {
-        real_cmd->microseconds += duration;
-        /* In case we are in the process of unblocking a client on keys,
-         * we are going to reprocess a command which was already process once so we do not want to account
-         * for the command twice */
-        if (!(c->flags & CLIENT_UNBLOCKED))
+        /* We want to increment the number of calls stats only once.
+         * why? because the blocked command can also be timedout or cancel, and we would still
+         * want to count for it. however we do want to accumulate the command processing time
+         * only after it is unblocked. */
+        if (!(c->flags & CLIENT_REPROCESSING_COMMAND))
             real_cmd->calls++;
-        /* If the client is blocked we will handle latency stats when it is unblocked. */
+
+        real_cmd->microseconds += duration;
```

Comment:
that is one option - personally I think that a better is to completely eliminate updateStatsOnUnblock. 
IMO a blocked client shuld not update stats and mostly act as if the command was never issued (aside for the monitor which is probably needed)

Reply:
done - please check if that was your intention. BTW - I can do the same for ACL check, but you mentioned that you think it is more correct this way...

Reply:
even if we keep the current behavior, won't it be cleaner to embed `updateStatsOnUnblock` in `call()`?
ohh, actually it may be an issue with modules and CLIENT UNBLOCK..

Reply:
I think we can close this sub thread